### PR TITLE
Crossreferences to standard library in mypy docs, part 3

### PR DESCRIPTION
**Pull request by hoefling**
Added references to the following documents:

* `error_code_list.rst`
* `error_code_list2.rst`
* `extending_mypy.rst`
* `faq.rst`
* `final_attrs.rst`
* `getting_started.rst`

This is part of splitting up the changes in #7624 into more readable PRs.

### CODE REVIEW

File: docs/source/error_code_list.rst [hunk 1]
```diff
@@ -415,11 +415,11 @@ Example:
     # Error: Dict entry 0 has incompatible type "str": "str"; expected "str": "int"  [dict-item]
     d: Dict[str, int] = {'key': 'value'}
 
-Check TypedDict items [typeddict-item]
---------------------------------------
+Check ``TypedDict`` items [typeddict-item]
```
Comment (hoefling) [inline docs/source/error_code_list.rst]: Please advise whether verbatim texts in section headers are useful. This renders as:

image

The side effect is also verbatim text in the table of contents:

image

File: docs/source/extending_mypy.rst [hunk 2]
```diff
@@ -185,12 +185,12 @@ For example in this code:
    x[0] = 42
 
 mypy will call ``get_method_signature_hook("ctypes.Array.__setitem__")``
-so that the plugin can mimic the ``ctypes`` auto-convert behavior.
+so that the plugin can mimic the :py:mod:`ctypes` auto-convert behavior.
 
-**get_attribute_hook** overrides instance member field lookups and property
+**get_attribute_hook()** overrides instance member field lookups and property
```
Comment (hoefling) [inline docs/source/extending_mypy.rst]: All other paragraphs use parentheses in hook names, I suppose they were omitted by mistake here.

File: docs/source/getting_started.rst [hunk 3]
```diff
@@ -160,16 +160,16 @@ Arguments with default values can be annotated like so:
...
+*********************
 
 So far, we've added type hints that use only basic concrete types like
 ``str`` and ``float``. What if we want to express more complex types,
 such as "a list of strings" or "an iterable of ints"?
 
-You can find many of these more complex static types inside of the ``typing``
+You can find many of these more complex static types inside of the :py:mod:`typing`
 module. For example, to indicate that some function can accept a list of
-strings, use the ``List`` type from the ``typing`` module:
```
Comment (hoefling) [inline docs/source/getting_started.rst]: Second change unrelated to referencing. IMO with the `List` reference, there's no need to mention the module again - it's also referenced one line above that. What do you think?

File: docs/source/faq.rst [hunk 4]
```diff
@@ -164,7 +163,7 @@ monkey patching of methods.
 How is mypy different from Cython?
 **********************************
 
-`Cython :doc:<cython:index>` is a variant of Python that supports
+:doc:`Cython <cython:index>` is a variant of Python that supports
```
Comment (hoefling) [inline docs/source/faq.rst]: Fixing my own errors here. No idea why Sphinx didn't catch this on build.

File: docs/source/extending_mypy.rst [hunk 5]
```diff
@@ -98,7 +98,7 @@ High-level overview
 *******************
 
 Every entry point function should accept a single string argument
-that is a full mypy version and return a subclass of ``mypy.plugins.Plugin``:
```
Comment (hoefling) [inline docs/source/extending_mypy.rst]: One of two changes in this PR that are not related to referencing, typo fix.

File: docs/source/error_code_list.rst (hunk 1 above)
Comment (gvanrossum) [reply docs/source/error_code_list.rst]: I think it looks better without the heavy formatting. TBH I find most CSS styling for code blocks over the top, and red is particularly bad in a heading. I also don't care about the box (or shading in other CSS setups).

File: docs/source/extending_mypy.rst (hunk 2 above)
Comment (gvanrossum) [reply docs/source/extending_mypy.rst]: +1

File: docs/source/getting_started.rst (hunk 3 above)
Comment (gvanrossum) [reply docs/source/getting_started.rst]: +1

(PS. When commenting on a diff chunk, please attach the comment to the last line, so the context in email or in the Conversation view is clearer. I think you can now also shift-click to attach a comment to a range of lines.)

File: docs/source/faq.rst (hunk 4 above)
Comment (gvanrossum) [reply docs/source/faq.rst]: Didn't you have to run it with special flags to check errors?

File: docs/source/error_code_list.rst (hunk 1 above)
Comment (hoefling) [reply docs/source/error_code_list.rst]: Agree. IMO the best solution is simply use monospaced font for verbatim formatting like Python stdlib docs do that (and afaik Javadoc too). If you don't mind, I will also replace other occurences of verbatim text in section titles in this PR.

File: docs/source/getting_started.rst (hunk 3 above)
Comment (hoefling) [reply docs/source/getting_started.rst]: Got it!

File: docs/source/faq.rst (hunk 4 above)
Comment (hoefling) [reply docs/source/faq.rst]: I did (using `sphinx-build -WEan`), however Sphinx ignores markup errors and just renders the text as-is. Neither does `doc8` nor `restructuredtext-lint`, sadly.
